codec-http2/src/test/java/io/netty/handler/codec/http2/Http2SettingsTest.java

```diff
@@ -75,4 +75,14 @@ public void settingsShouldSupportUnsignedShort() {
         settings.put(key, 123L);
         assertEquals(123L, (long) settings.get(key));
     }
+
+    @Test
+    public void boundarySettingsShouldBeSet() {
+        settings.maxHeaderListSize(Integer.MAX_VALUE + 1);
```

Comment:
Will this cause a compiler warning? Maybe use bit arithmetic so that it doesn't know we're purposely overflowing the integer? Or set the `long` value directly using `put` rather than calling `maxHeaderListSize`?